src/core/mac/mac.cpp

```diff
@@ -165,6 +165,11 @@ Mac::Mac(ThreadNetif &aThreadNetif):
     mTxFrame = static_cast<Frame *>(otPlatRadioGetTransmitBuffer(mNetif.GetInstance()));
 }
 
+void Mac::PrepareRadio()
+{
+    otPlatRadioSetCallbacks(mNetif.GetInstance(), ReceiveDoneTask, TransmitDoneTask);
```

Comment:
@nibanks Could you provide more detail on the use cases here?  Enabling a configuration where the stack runs primarily on the host makes sense to me.  Let's call such a configuration a "transceiver" configuration (OT on host) as opposed to the more classic "NCP" (OT on coprocessor) configuration.  However, I don't see why one would ever need to dynamically switch between transceiver and NCP configurations, as you'll want OpenThread state to live in one place -- either on the host or the NCP.  It'd be good to understand the use cases that are driving "dynamic switching" more to assess whether they are worth the added complexity.